src/main/java/org/jitsi/jigasi/stats/StatsHandler.java

```diff
@@ -0,0 +1,377 @@
+/*
+ * Jigasi, the JItsi GAteway to SIP.
+ *
+ * Copyright @ 2015 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.jigasi.stats;
+
+import net.java.sip.communicator.service.protocol.*;
+import net.java.sip.communicator.service.protocol.event.*;
+import net.java.sip.communicator.util.*;
+import org.jitsi.jigasi.*;
+import org.jitsi.jigasi.util.*;
+import org.jitsi.stats.media.*;
+import org.jitsi.util.concurrent.*;
+import org.osgi.framework.*;
+
+import java.util.*;
+import java.util.concurrent.*;
+
+/**
+ * The entry point for stats logging.
+ * Listens for start of calls to start reporting and end of calls
+ * to stop reporting.
+ *
+ * @author Damian Minkov
+ */
+public class StatsHandler
+    extends CallChangeAdapter
+{
+    /**
+     * The logger.
+     */
+    private final static Logger logger = Logger.getLogger(StatsHandler.class);
+
+    /**
+     * CallStats account property: appId
+     * The account id from callstats.io settings.
+     */
+    private static final String CS_ACC_PROP_APP_ID = "CallStats.appId";
+
+    /**
+     * CallStats account property: keyId
+     * The key id from callstats.io settings.
+     */
+    private static final String CS_ACC_PROP_KEY_ID = "CallStats.keyId";
+
+    /**
+     * CallStats account property: keyPath
+     * The path to the key file on the file system.
+     */
+    private static final String CS_ACC_PROP_KEY_PATH = "CallStats.keyPath";
+
+    /**
+     * CallStats account property: conferenceIDPrefix
+     * The domain that this stats account is handling.
+     */
+    private static final String CS_ACC_PROP_CONFERENCE_PREFIX
+        = "CallStats.conferenceIDPrefix";
+
+    /**
+     * CallStats account property: STATISTICS_INTERVAL
+     * The interval on which to publish statistics.
+     */
+    private static final String CS_ACC_PROP_STATISTICS_INTERVAL
+        = "CallStats.STATISTICS_INTERVAL";
+
+    /**
+     * The default value for statistics interval.
+     */
+    public static final int DEFAULT_STAT_INTERVAL = 5000;
+
+    /**
+     * CallStats account property: jigasiId
+     * The jigasi id to report to callstats.io.
+     */
+    private static final String CS_ACC_PROP_JIGASI_ID
+        = "CallStats.jigasiId";
+
+    /**
+     * The default jigasi id to use if setting is missing.
+     */
+    public static final String DEFAULT_JIGASI_ID = "jigasi";
```

Comment:
I wonder if we can move most of these (as well as their counterparts in jitsi-videobridge) to a common abstract class in jitsi-stats.